fdb-record-layer-geophile/src/main/java/com/apple/foundationdb/record/geophile/GeophileSpatial.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * GeophileSpatial.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.geophile;
+
+import com.apple.foundationdb.annotation.API;
+import com.geophile.z.Space;
+import com.geophile.z.SpatialObject;
+import com.geophile.z.spatialobject.jts.JTS;
+import com.geophile.z.spatialobject.jts.JTSSpatialObject;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.CoordinateSequenceFilter;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+import org.locationtech.jts.io.WKBWriter;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.io.WKTWriter;
+import org.locationtech.jts.io.geojson.GeoJsonReader;
+import org.locationtech.jts.io.geojson.GeoJsonWriter;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Helper methods for interfacing with Geophile, specifically for implementing geospatial with lat, lon coordinates.
+ */
+@API(API.Status.EXPERIMENTAL)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")    // Allow WKB, WKT
+public class GeophileSpatial {
+    /*
+      The lat/lon coordinate system is
+
+      - latitude: -90.0 to +90.0
+      - longitude: -180.0 to 180.0 with wraparound
+
+      The interleave pattern is [lon, lat, lon, lat, ..., lon], reflecting the fact that longitude covers a numeric range
+      twice that of latitude.
+    */
+
+    public static final int LAT_LON_DIMENSIONS = 2;
+    public static final double MIN_LAT = -90;
+    public static final double MAX_LAT = 90;
+    public static final double MIN_LON = -180;
+    public static final double MAX_LON = 180;
+    private static final int LAT_BITS = 28;
+    private static final int LON_BITS = 29;
+
+    private GeophileSpatial() {
+    }
+
+    static {
+        SLF4JBridgeHandler.removeHandlersForRootLogger();
+        SLF4JBridgeHandler.install();
+        java.util.logging.Logger.getLogger("").setLevel(java.util.logging.Level.FINEST);
```

Comment:
The idea is to have the record layer's log setup do all the level filtering by passing everything through `java.util.logging`.

Reply:
I see. I guess it's a little unfortunate that if the record layer's log level is, say, `INFO`, and there's a conditional `FINE` j.u.l. log message in geophile, then I think it will go ahead and create the log message only for it to be filtered out later (if I understand how logging works). I guess the alternative would be to set the root level to the log level of, like, `getLogger(GeophileSpatial.class)` (after translating it to it's j.u.l. equivalent), but maybe that's not worth it. Not sure how chatty these logs are.